File: problems/easy/Excel_Sheet_Column_Number.rb
# Given a column title as appear in an Excel sheet, return its corresponding column number.
# 
# For example:
# 
#     A -> 1
#     B -> 2
#     C -> 3
#     ...
#     Z -> 26
#     AA -> 27
#     AB -> 28 
#     ...
# Example 1:
# 
# Input: "A"
# Output: 1
# Example 2:
# 
# Input: "AB"
# Output: 28
# Example 3:
# 
# Input: "ZY"
# Output: 701
# https://leetcode.com/problems/excel-sheet-column-number/description/

# @param {String} s
# @return {Integer}
def title_to_number(s)
  return 0 if s.nil?
  length = s.length
  i = 0
  num = 0
  while(i<length)
    num += (s[i].ord - 64)* (26**(length - i - 1))
    i += 1
  end
  num
end

File: practice/2023/Excel_Sheet_Column_Number.go
/*
 * @Author: hongwei.sun
 * @Date: 2023-10-04 16:13:12
 * @LastEditors: your name
 * @LastEditTime: 2023-10-04 16:59:43
 * @Description: file content
 */

//  https://leetcode.com/problems/excel-sheet-column-number/description/
//  it's one 26 system convert question
func titleToNumber(columnTitle string) int {
	res := 0
	scale := 1
	for i := len(columnTitle) - 1; i >= 0; i -= 1 {
		res += (int(columnTitle[i]) - int('A') + 1) * scale
		scale *= 26
	}
	return res
}
